packages/kbn-securitysolution-io-ts-alerting-types/src/actions/index.ts

```diff
@@ -18,35 +18,65 @@ export const RuleActionId = t.string;
 export type RuleActionTypeId = t.TypeOf<typeof RuleActionTypeId>;
 export const RuleActionTypeId = t.string;
 
+export type RuleActionUuid = t.TypeOf<typeof RuleActionUuid>;
+export const RuleActionUuid = t.string;
+
+export type RuleActionOptionalUuid = t.TypeOf<typeof RuleActionOptionalUuid>;
+export const RuleActionOptionalUuid = t.union([RuleActionUuid, t.undefined]);
+
 /**
  * Params is an "object", since it is a type of RuleActionParams which is action templates.
  * @see x-pack/plugins/alerting/common/rule.ts
  */
 export type RuleActionParams = t.TypeOf<typeof RuleActionParams>;
 export const RuleActionParams = saved_object_attributes;
 
+const BaseRuleAction = t.type({
+  group: RuleActionGroup,
+  id: RuleActionId,
+  action_type_id: RuleActionTypeId,
+  params: RuleActionParams,
+});
+
 export type RuleAction = t.TypeOf<typeof RuleAction>;
 export const RuleAction = t.exact(
-  t.type({
-    group: RuleActionGroup,
-    id: RuleActionId,
-    action_type_id: RuleActionTypeId,
-    params: RuleActionParams,
-  })
+  t.intersection([BaseRuleAction, t.type({ uuid: RuleActionUuid })])
 );
 
+export type RuleActionWithoutUuid = t.TypeOf<typeof BaseRuleAction>;
+
 export type RuleActionArray = t.TypeOf<typeof RuleActionArray>;
 export const RuleActionArray = t.array(RuleAction);
 
+// --------
+
+const BaseRuleActionCamel = t.type({
+  group: RuleActionGroup,
+  id: RuleActionId,
+  actionTypeId: RuleActionTypeId,
+  params: RuleActionParams,
+});
+
 export type RuleActionCamel = t.TypeOf<typeof RuleActionCamel>;
 export const RuleActionCamel = t.exact(
-  t.type({
-    group: RuleActionGroup,
-    id: RuleActionId,
-    actionTypeId: RuleActionTypeId,
-    params: RuleActionParams,
-  })
+  t.intersection([BaseRuleActionCamel, t.type({ uuid: RuleActionUuid })])
 );
 
 export type RuleActionArrayCamel = t.TypeOf<typeof RuleActionArrayCamel>;
 export const RuleActionArrayCamel = t.array(RuleActionCamel);
+
+// --------
+
+export const RuleActionWithOptionalUuidArray = t.array(
+  t.exact(t.intersection([BaseRuleAction, t.partial({ uuid: RuleActionOptionalUuid })]))
+);
+
+export const RuleActionWithOptionalUuidArrayCamel = t.array(
+  t.exact(t.intersection([BaseRuleActionCamel, t.partial({ uuid: RuleActionOptionalUuid })]))
+);
+
+// --------
+export type RuleActionWithoutUuidArray = t.TypeOf<typeof RuleActionWithoutUuidArray>;
+export const RuleActionWithoutUuidArray = t.array(t.exact(BaseRuleAction));
+
+export const RuleActionWithoutUuidArrayCamel = t.array(t.exact(BaseRuleActionCamel));
```

Comment:
I had to modify these schemas to have:
An interface for create API without uuid field in the RuleActions
An interface for the update API with an optional uuid field in the RuleActions

And their camelCase versions.